tactics/class_tactics.ml

```diff
@@ -1003,7 +1054,15 @@ let typeclasses_eauto ?(only_classes=false) ?(st=TransparentState.full)
   let modes = List.map Hint_db.modes dbs in
   let modes = List.fold_left (GlobRef.Map.union (fun _ m1 m2 -> Some (m1@m2))) GlobRef.Map.empty modes in
   let depth = match depth with None -> get_typeclasses_depth () | Some l -> Some l in
-  Search.eauto_tac (modes,st) ~only_classes ?strategy ~depth ~dep:true dbs
+  Proofview.tclBIND
+    (Search.eauto_tac (modes,st) ~only_classes ?strategy ~depth ~dep:true dbs)
+    (* Stuck goals will remain here, we could also shelve them, but this way
+       `solve [typeclasses eauto]` can check there are no stuck goals
+       remaining. *)
+    (fun l -> Proofview.tclUNIT ())
```

Comment:
`tclBIND foo (fun _ -> tclUNIT ())` is just `tclIGNORE foo` I think

Reply:
Indeed